packages/design-system/src/components/Table/TableCell.test.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import Table from './Table';
+import TableBody from './TableBody';
+import TableCell from './TableCell';
+import TableHead from './TableHead';
+import TableRow from './TableRow';
+import { mount } from 'enzyme';
+
+const defaultTableHeadChildren = (
+  <TableRow key="1">
+    <TableCell>Column a</TableCell>
+    <TableCell>Column b</TableCell>
+  </TableRow>
+);
+const defaultTableHeadProps = {
+  className: 'foo-head',
+};
+
+const defaultTableBodyChildren = (
+  <TableRow key="2">
+    <TableCell component="th">Cell a</TableCell>
+    <TableCell>Cell b</TableCell>
+  </TableRow>
+);
+const defaultTableBodyProps = {
+  className: 'foo-body',
+};
+
+function renderHead(customProps = {}, children) {
+  const props = Object.assign({}, customProps);
+
+  if (!children) {
+    children = <TableHead {...defaultTableHeadProps}>{defaultTableHeadChildren}</TableHead>;
+  }
+
+  return {
+    props: props,
+    wrapper: mount(<Table {...props}>{children}</Table>),
+  };
+}
+
+function renderBody(customProps = {}, children) {
+  const props = Object.assign({}, customProps);
+
+  if (!children) {
+    children = <TableBody {...defaultTableBodyProps}>{defaultTableBodyChildren}</TableBody>;
+  }
+
+  return {
+    props: props,
+    wrapper: mount(<Table {...props}>{children}</Table>),
+  };
+}
+
+describe('Table', function () {
+  it('renders a table thead <th> element', () => {
```

Comment:
Could we also add some tests for using the `component` and `scope` props, which overrides the default values based off the parent element

Reply:
Yes, already in the test (line 100 to 122) but I'll modified the text to say `renders a table tbody <th> row header element which overwrites default header row component to <th>` and `sets a table tbody <th> role="rowheader" which overwrites default role value "cell"`

Reply:
Yeah i think this test file would be good to revisit, I'm having some trouble understanding them, as I dont see the `component` and `scope` props being explicitly used or passed in